src/test/java/com/fasterxml/jackson/databind/tofix/ShapeArrayWithAnyGetter4961Test.java

```diff
@@ -0,0 +1,84 @@
+package com.fasterxml.jackson.databind.tofix;
+
+import java.util.Map;
+import java.util.TreeMap;
+
+import org.junit.jupiter.api.Test;
+
+import com.fasterxml.jackson.annotation.*;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.testutil.DatabindTestUtil;
+import com.fasterxml.jackson.databind.testutil.failure.JacksonTestFailureExpected;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+// [databind#4961] Serialization for `JsonFormat.Shape.ARRAY` does not work when there is `@JsonAnyGetter`
+public class ShapeArrayWithAnyGetter4961Test
+    extends DatabindTestUtil
+{
+
+    static class WrapperForAnyGetter {
+        public BeanWithAnyGetter value;
+    }
+
+    @JsonPropertyOrder({ "firstProperty", "secondProperties", "anyProperty", "forthProperty" })
+    @JsonFormat(shape = JsonFormat.Shape.ARRAY)
+    static class BeanWithAnyGetter {
+        public String firstProperty = "first";
+        public String secondProperties = "second";
+        public String forthProperty = "forth";
+        @JsonAnyGetter
+        public Map<String, String> getAnyProperty() {
+            Map<String, String> map = new TreeMap<>();
+            map.put("third_A", "third_A");
+            map.put("third_B", "third_B");
+            return map;
+        }
+    }
+
+    final ObjectMapper MAPPER = newJsonMapper();
+
+    @JacksonTestFailureExpected
```

Comment:
TODO: Move to test dir, out of tofix

Reply:
Is the test then not failing or... ?

Reply:
Since the test is expected behavior, we may move to tests dir.